src/mrack/providers/openstack.py

```diff
@@ -441,6 +458,7 @@ async def wait_till_provisioned(
                 if error_attempts > SERVER_ERROR_RETRY:
                     raise ProvisioningError(uuid)
 
+            poll_sleep += 0.5  # increase delays to check the longer it takes
```

Comment:
Do you think it would be useful to add this?
```python
logger.debug(f"Sleeping for {poll_sleep} seconds")
```

Reply:
Initial I thought it would be too noisy, but seems bearable.  Example of run:

```
2021-02-07 18:35:46,550 mrack.providers.provider INFO OpenStack: Issuing provisioning of 9 hosts
2021-02-07 18:35:46,550 mrack.providers.openstack INFO OpenStack: Creating server master.mrack.test
2021-02-07 18:35:46,550 mrack.providers.openstack INFO OpenStack: Creating server replica.mrack.test
2021-02-07 18:35:46,551 mrack.providers.openstack INFO OpenStack: Creating server client.mrack.test
2021-02-07 18:35:46,551 mrack.providers.openstack INFO OpenStack: Creating server client2.mrack.test
2021-02-07 18:35:46,552 mrack.providers.openstack INFO OpenStack: Creating server client3.mrack.test
2021-02-07 18:35:46,552 mrack.providers.openstack INFO OpenStack: Creating server client4.mrack.test
2021-02-07 18:35:46,552 mrack.providers.openstack INFO OpenStack: Creating server client5.mrack.test
2021-02-07 18:35:46,553 mrack.providers.openstack INFO OpenStack: Creating server client6.mrack.test
2021-02-07 18:35:46,553 mrack.providers.openstack INFO OpenStack: Creating server client7.mrack.test
2021-02-07 18:35:50,043 mrack.providers.provider INFO OpenStack: Provisioning issued
2021-02-07 18:35:50,043 mrack.providers.provider INFO OpenStack: Waiting for all hosts to be available
2021-02-07 18:35:50,044 mrack.providers.openstack DEBUG bc4e2be0-2ad6-4c56-93c8-4812cf539f70: sleeping for 36.410746682643676 seconds
2021-02-07 18:35:50,045 mrack.providers.openstack DEBUG 9908fe32-74c4-4978-9fdf-01ec36a6dfbb: sleeping for 35.366141324753414 seconds
2021-02-07 18:35:50,045 mrack.providers.openstack DEBUG a9d2773a-82dd-4d18-9026-a26c6c70da08: sleeping for 22.213545260895046 seconds
2021-02-07 18:35:50,045 mrack.providers.openstack DEBUG 9244a0d6-2be9-490b-b9c1-33be8709f123: sleeping for 24.272930020442956 seconds
2021-02-07 18:35:50,046 mrack.providers.openstack DEBUG 255664d7-e1e2-48a7-ad8c-b64d10d3a53d: sleeping for 29.100387340669513 seconds
2021-02-07 18:35:50,046 mrack.providers.openstack DEBUG 39215daa-020b-4c68-8ec0-4ce1543f4688: sleeping for 20.907414711579754 seconds
2021-02-07 18:35:50,046 mrack.providers.openstack DEBUG 4768efa0-bc35-4ea8-91fc-fbaa6bde8e4f: sleeping for 12.699270548378916 seconds
2021-02-07 18:35:50,047 mrack.providers.openstack DEBUG 92a1e456-3232-4894-b2f9-dff9f5569660: sleeping for 21.521139682535193 seconds
2021-02-07 18:35:50,047 mrack.providers.openstack DEBUG b5ef858c-a059-42a0-8fe8-2984dd746d9a: sleeping for 33.517761504283214 seconds
2021-02-07 18:36:02,759 mrack.providers.openstack DEBUG Waiting for: 4768efa0-bc35-4ea8-91fc-fbaa6bde8e4f
2021-02-07 18:36:03,560 mrack.providers.openstack DEBUG 4768efa0-bc35-4ea8-91fc-fbaa6bde8e4f: sleeping for 9.48 seconds
2021-02-07 18:36:10,961 mrack.providers.openstack DEBUG Waiting for: 39215daa-020b-4c68-8ec0-4ce1543f4688
2021-02-07 18:36:11,569 mrack.providers.openstack DEBUG Waiting for: 92a1e456-3232-4894-b2f9-dff9f5569660
2021-02-07 18:36:11,756 mrack.providers.openstack INFO OpenStack: Host 39215daa-020b-4c68-8ec0-4ce1543f4688 was provisioned in 21.7s
2021-02-07 18:36:12,259 mrack.providers.openstack DEBUG Waiting for: a9d2773a-82dd-4d18-9026-a26c6c70da08
2021-02-07 18:36:13,055 mrack.providers.openstack DEBUG a9d2773a-82dd-4d18-9026-a26c6c70da08: sleeping for 9.48 seconds
2021-02-07 18:36:13,137 mrack.providers.openstack INFO OpenStack: Host 92a1e456-3232-4894-b2f9-dff9f5569660 was provisioned in 23.1s
2021-02-07 18:36:14,309 mrack.providers.openstack DEBUG 4768efa0-bc35-4ea8-91fc-fbaa6bde8e4f: sleeping for 9.98 seconds
2021-02-07 18:36:14,320 mrack.providers.openstack DEBUG Waiting for: 9244a0d6-2be9-490b-b9c1-33be8709f123
2021-02-07 18:36:15,129 mrack.providers.openstack INFO OpenStack: Host 9244a0d6-2be9-490b-b9c1-33be8709f123 was provisioned in 25.1s
2021-02-07 18:36:19,150 mrack.providers.openstack DEBUG Waiting for: 255664d7-e1e2-48a7-ad8c-b64d10d3a53d
2021-02-07 18:36:19,945 mrack.providers.openstack INFO OpenStack: Host 255664d7-e1e2-48a7-ad8c-b64d10d3a53d was provisioned in 29.9s
2021-02-07 18:36:23,426 mrack.providers.openstack DEBUG a9d2773a-82dd-4d18-9026-a26c6c70da08: sleeping for 9.98 seconds
2021-02-07 18:36:23,566 mrack.providers.openstack DEBUG Waiting for: b5ef858c-a059-42a0-8fe8-2984dd746d9a
2021-02-07 18:36:24,450 mrack.providers.openstack INFO OpenStack: Host b5ef858c-a059-42a0-8fe8-2984dd746d9a was provisioned in 34.4s
2021-02-07 18:36:25,412 mrack.providers.openstack DEBUG Waiting for: 9908fe32-74c4-4978-9fdf-01ec36a6dfbb
2021-02-07 18:36:25,938 mrack.providers.openstack DEBUG 4768efa0-bc35-4ea8-91fc-fbaa6bde8e4f: sleeping for 10.48 seconds
2021-02-07 18:36:26,456 mrack.providers.openstack DEBUG Waiting for: bc4e2be0-2ad6-4c56-93c8-4812cf539f70
2021-02-07 18:36:26,495 mrack.providers.openstack DEBUG 9908fe32-74c4-4978-9fdf-01ec36a6dfbb: sleeping for 9.48 seconds
2021-02-07 18:36:27,422 mrack.providers.openstack DEBUG bc4e2be0-2ad6-4c56-93c8-4812cf539f70: sleeping for 9.48 seconds
2021-02-07 18:36:34,588 mrack.providers.openstack DEBUG a9d2773a-82dd-4d18-9026-a26c6c70da08: sleeping for 10.48 seconds
2021-02-07 18:36:36,948 mrack.providers.openstack DEBUG 9908fe32-74c4-4978-9fdf-01ec36a6dfbb: sleeping for 9.98 seconds
2021-02-07 18:36:38,070 mrack.providers.openstack DEBUG 4768efa0-bc35-4ea8-91fc-fbaa6bde8e4f: sleeping for 10.98 seconds
2021-02-07 18:36:38,086 mrack.providers.openstack DEBUG bc4e2be0-2ad6-4c56-93c8-4812cf539f70: sleeping for 9.98 seconds
2021-02-07 18:36:45,954 mrack.providers.openstack DEBUG a9d2773a-82dd-4d18-9026-a26c6c70da08: sleeping for 10.98 seconds
2021-02-07 18:36:49,027 mrack.providers.openstack DEBUG bc4e2be0-2ad6-4c56-93c8-4812cf539f70: sleeping for 10.48 seconds
2021-02-07 18:36:49,352 mrack.providers.openstack DEBUG 9908fe32-74c4-4978-9fdf-01ec36a6dfbb: sleeping for 10.48 seconds
2021-02-07 18:36:49,748 mrack.providers.openstack DEBUG 4768efa0-bc35-4ea8-91fc-fbaa6bde8e4f: sleeping for 11.48 seconds
2021-02-07 18:36:58,510 mrack.providers.openstack DEBUG a9d2773a-82dd-4d18-9026-a26c6c70da08: sleeping for 11.48 seconds
2021-02-07 18:37:00,326 mrack.providers.openstack INFO OpenStack: Host bc4e2be0-2ad6-4c56-93c8-4812cf539f70 was provisioned in 70.3s
2021-02-07 18:37:00,902 mrack.providers.openstack DEBUG 9908fe32-74c4-4978-9fdf-01ec36a6dfbb: sleeping for 10.98 seconds
2021-02-07 18:37:02,237 mrack.providers.openstack INFO OpenStack: Host 4768efa0-bc35-4ea8-91fc-fbaa6bde8e4f was provisioned in 72.2s
2021-02-07 18:37:10,938 mrack.providers.openstack INFO OpenStack: Host a9d2773a-82dd-4d18-9026-a26c6c70da08 was provisioned in 80.9s
2021-02-07 18:37:13,009 mrack.providers.openstack DEBUG 9908fe32-74c4-4978-9fdf-01ec36a6dfbb: sleeping for 11.48 seconds
2021-02-07 18:37:25,380 mrack.providers.openstack DEBUG 9908fe32-74c4-4978-9fdf-01ec36a6dfbb: sleeping for 11.98 seconds
2021-02-07 18:37:38,179 mrack.providers.openstack INFO OpenStack: Host 9908fe32-74c4-4978-9fdf-01ec36a6dfbb was provisioned in 108.1s
2021-02-07 18:37:38,180 mrack.providers.provider INFO OpenStack: All hosts reached provisioning final state (ACTIVE or ERROR)
2021-02-07 18:37:38,180 mrack.providers.provider INFO OpenStack: Provisioning duration: 0:01:51.630257
```